src/components/images/ImageEdit.js

```diff
@@ -1,84 +1,62 @@
 // @flow
-import React, { Component } from 'react';
+import React, { useState } from 'react';
+import { useMedia } from 'react-use';
 import { Redirect } from 'react-router-dom';
 import { IoIosArrowRoundBack } from 'react-icons/io';
 import ImageService from '../../services/ImageService';
 import ActivityInidicator from '../utility/ActivityIndicator';
+import MobileTagging from '../tagging/MobileTagging';
 import SingleTagging from '../tagging/SingleTagging';
+import useTaggableImage from '../../hooks/useTaggableImage';
 import { SaveButton, ButtonLink } from '../utility/buttons';
 import Toolbar from '../utility/Toolbar';
-import type { Image } from '../../models/Image';
-import type { Tag } from '../../models/Tag';
+import type { TaggableImage } from '../../hooks/useTaggableImage';
 
 type Props = {
   match: {
     params: { id: string },
   },
 };
 
-type State = {
-  image: ?Image,
-  isSubmitted: boolean,
-};
-
-class ImageEdit extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
-      image: null,
-      isSubmitted: false,
-    };
-  }
-
-  async componentDidMount() {
-    const image = await ImageService.get(this.props.match.params.id);
-    this.setState({ image });
-  }
+function ImageEdit(props: Props) {
+  const isMobile = useMedia('(max-width: 600px)');
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const image: ?TaggableImage = useTaggableImage(props.match.params.id);
```

Comment:
This is pretty cool!

Reply:
I wonder how it works when you keep adding aspects to it though, will we just have millions of hooks?
- taggableImage
- saveableImage
- deletableImage
- authorizableImage
?

That doesn't seem very feasible, but since we can parametrize the call we could do something like:

```
useImage(['taggable', 'deletable', 'authorizable' ])
```

and add the functionality dynamically right?

Reply:
Actually this is going to mess with types again as well :/ since you have no way of specifying what functionality is actually provided depending on the input...

Reply:
Hmmm maybe we'll find a clever solution...